### PR TITLE
GH-1319: Order Lib Doc

### PR DESCRIPTION
**Pull request by plabadie**

### CODE REVIEW

File: _pages/order-lib.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+title: Order Lib Release Notes
+---
+
+Spartacus 4.2 introduces the Order Library. This is the latest iteration of the code splitting initiative. The main goal of code splitting is to reduce the initial download size of a Spartacus enabled app by leveraging lazy loading of feature libraries.
+
+# Backwards compatibility for existing apps by default
+
+By default, if you upgrade to 4.2 from 4.x, the order library will not be used. Instead, your app will continue to use the old/current order related code. The non-lib ordr code is still present in Spartacus and will onoly be removed in 5.0. This allows Spartacus 4.2 to be backwards compatible with previous 4.x versions.
```
Comment (WeizhengSap) [inline _pages/order-lib.md:9]: non-lib `ordr` code (typo)

File: _pages/order-lib.md [hunk 2]
```diff
@@ -4,19 +4,46 @@ title: Order Lib Release Notes
 
 Spartacus 4.2 introduces the Order Library. This is the latest iteration of the code splitting initiative. The main goal of code splitting is to reduce the initial download size of a Spartacus enabled app by leveraging lazy loading of feature libraries.
 
-# Backwards compatibility for existing apps by default
+# Existing Apps upgrading to 4.2: keep backwards compatibility
 
-By default, if you upgrade to 4.2 from 4.x, the order library will not be used. Instead, your app will continue to use the old/current order related code. The non-lib order code is still present in Spartacus and will only be removed in 5.0. This allows Spartacus 4.2 to be backwards compatible with previous 4.x versions.
+The the new Order Library is not be backwards compatible. Therefore, by default, if you upgrade to 4.2 from 4.x, the order library will NOT be used. Instead, your app will continue to use the old/current order related code to preserve the backwards coompatibility of the upgrade. The non-lib order code is still present in Spartacus. This allows Spartacus 4.2 to be backwards compatible with previous 4.x versions.
```
Comment (WeizhengSap) [inline _pages/order-lib.md:9]: 2 `the`

File: _pages/order-lib.md [hunk 3]
```diff
@@ -4,19 +4,46 @@ title: Order Lib Release Notes
...
 
 # Order lib used in new Spartcus Apps.
 
-In Spartacus 4.2, if you generate a new Spartacuus app with the schematics, the app will use the new Order Library.
+In Spartacus 4.2, if you generate a new Spartacus app with the schematics, the app will use the new Order Library.
+
+# Order lib used in new Spartcus Apps.
+
+The the new Order Library is not be backwards compatible. Some changes to the code were required to make the order related features work within the order library. If you have custom code iin your app, theere is a chance that this code will require changees for it to work with the oorder library. If you still want to explore the possibility of using the order lib in an existing app upgraded to 4.2, you can switch to use the cart lib by changing the imports. ( in a standard app structure, these imports are in )
```
Comment (WeizhengSap) [inline _pages/order-lib.md:17]: `you can switch to use the cart lib` should be `you can switch to use the order lib`

File: _pages/order-lib.md [hunk 4]
```diff
@@ -4,19 +4,46 @@ title: Order Lib Release Notes
...
+- OrderCancellationModule (@spartacus/storefront)
+- OrderDetailsModule (@spartacus/storefront)
+- OrderHistoryModule (@spartacus/storefront)
+- OrderReturnModule (@spartacus/storefront)
+- ReplenishmentOrderDetailsModule (@spartacus/storefront)
+- ReplenishmentOrderHistoryModule (@spartacus/storefront)
+- ReturnRequestDetailModule (@spartacus/storefront)
+- ReturnRequestListModule (@spartacus/storefront)
+
+The add the modules to use the new order lib:
```
Comment (WeizhengSap) [inline _pages/order-lib.md:32]: Then

File: _pages/order-lib.md [hunk 5]
```diff
@@ -4,19 +4,46 @@ title: Order Lib Release Notes
...
 
-By default, if you upgrade to 4.2 from 4.x, the order library will not be used. Instead, your app will continue to use the old/current order related code. The non-lib order code is still present in Spartacus and will only be removed in 5.0. This allows Spartacus 4.2 to be backwards compatible with previous 4.x versions.
+The the new Order Library is not be backwards compatible. Therefore, by default, if you upgrade to 4.2 from 4.x, the order library will NOT be used. Instead, your app will continue to use the old/current order related code to preserve the backwards coompatibility of the upgrade. The non-lib order code is still present in Spartacus. This allows Spartacus 4.2 to be backwards compatible with previous 4.x versions.
 
 # Order lib used in new Spartcus Apps.
 
-In Spartacus 4.2, if you generate a new Spartacuus app with the schematics, the app will use the new Order Library.
+In Spartacus 4.2, if you generate a new Spartacus app with the schematics, the app will use the new Order Library.
+
+# Order lib used in new Spartcus Apps.
```
Comment (WeizhengSap) [inline _pages/order-lib.md:15]: The title is not correct, we are talking about upgrade, not in new app.

File: _pages/order-lib.md [hunk 6]
```diff
@@ -4,19 +4,46 @@ title: Order Lib Release Notes
...
+
+The the new Order Library is not be backwards compatible. Some changes to the code were required to make the order related features work within the order library. If you have custom code iin your app, theere is a chance that this code will require changees for it to work with the oorder library. If you still want to explore the possibility of using the order lib in an existing app upgraded to 4.2, you can switch to use the cart lib by changing the imports. ( in a standard app structure, these imports are in )
+
+You first need to remove the imports of the order code that was moved.
```
Comment (WeizhengSap) [inline _pages/order-lib.md:19]: customers may not just import modules, there are also other stuff (models...). I don't know whether we need give these detailed stuff.

File: _pages/order-lib.md (hunk 5 above)
Comment (plabadie) [reply _pages/order-lib.md:15]: We're covering the 4.2 release.  Upgrading to 4.2 and starting a new app with 4.2 are 2 valid use cases, no?
Order lib is the default for the new app use case, while the legacy order code is the default for the upgrade.

File: _pages/order-lib.md (hunk 2 above)
Comment (plabadie) [reply _pages/order-lib.md:9]: Fixed

File: _pages/order-lib.md (hunk 3 above)
Comment (plabadie) [reply _pages/order-lib.md:17]: Indeed. 

Fixed

File: _pages/order-lib.md (hunk 4 above)
Comment (plabadie) [reply _pages/order-lib.md:32]: I removed this section

**Review by gladius-mtl — APPROVED** [review]

**Review by gladius-mtl — APPROVED** [review]

**Pull request merged**